android/src/main/java/com/reactcommunity/rndatetimepicker/RNDatePickerDialogModule.java

```diff
@@ -84,6 +84,25 @@ public void onClick(DialogInterface dialog, int which) {
     }
   }
 
+  @ReactMethod
+  public void close(Promise promise) {
+    FragmentActivity activity = (FragmentActivity) getCurrentActivity();
+    if (activity == null) {
+      promise.reject(
+              RNConstants.ERROR_NO_ACTIVITY,
+              "Tried to close a DatePicker dialog while not attached to an Activity");
```

Comment:
Hmm.. This might cause troubles with racy behavior. The app might lag out or the picker could be closed "at the same time" as the component dismounts, if so we are faced with a crash.

Maybe just catch and ignore the error on call site?

Reply:
I copied this behavior from the open-code, so i assume it's as equally safe as there, at least.

Reply:
I'm thinking that having it have the same guards as the open-functions will have it be merged easier upstream.

Reply:
Yeah I just meant in the `useEffect` we could catch the error message and ignore it, since our end goal is a closed date picker, and if the date picker isn't there we've achieved our goal

Reply:
If it crashes due to uncaught promise, we should be able to pick it up in local/dev, and iron out the kinks.

Reply:
> I'm thinking that having it have the same guards as the open-functions will have it be merged easier upstream.

Yeah that's a good point. Alright let's go with this.

Reply:
Can we catch this in our code somehow?

Reply:
in the js part of this lib where i call .close() we could always .catch(noop), as we don't actually care about the error.